### DISCUSSION TITLE
Updated to 0.58.5, Glorytun keeps losing connection

### DISCUSSION THREAD

**Discussion by bdruth** [Ysurac/openmptcprouter#2059] [Q&A]
How can I debug/diagnose what might be going on in 0.58.5 w/ Glorytun? It keeps restarting - client side says `Interface 'omrvpn' has link connectivity loss` - server side at the same timestamp reads
```
Sep  4 14:51:07 debian-09 glorytun-tcp-run[6407]: 206.214.228.4.10537: connected
Sep  4 14:51:10 debian-09 OMR-Service: No answer from VPN client end, restart Glorytun-TCP
```
Stats from OMR indicate that there was already traffic over `tun0` in the preceding minute ... 
<img width="662" alt="image" src="https://user-images.githubusercontent.com/444571/132106955-0a98ad9b-70c4-4ef8-a87a-2356bb502b2e.png">

No errors on any interfaces that I can see. I updated OMR last night and this appears to have been happening quite a bit since then. Just in the last 4 hrs, 4 times that I can see from the OMR client log buffer, the server says it's restarted glorytun 9 times.

Everything's up-to-date, server & OMR side.

**Comment by Ysurac** [comment]
On the VPS side there is a service (called "omr") that is running and try to ping router side, if no answer then VPN is restarted.
You can try to ping 10.255.255.2 from VPS and check if you get an answer.

**Reply by bdruth** [reply]
Yeah, so - in digging more, there's something whacky going on w/ Glorytun, I think. The individual WAN connections are fine, as near as ping can tell, but the Glorytun interface is another matter entirely.

<img width="1552" alt="omr_0 58 5_pings" src="https://user-images.githubusercontent.com/444571/132108559-a384762b-5631-400a-a8a4-396c370a5dc3.png">

| col1 | col2 |
| ---- | ---- |
| ping via wan1<br>to server-ip |ping via wan2<br>to server-ip |
| ping from lan<br>to server-ip |ping via tun0<br>to 10.255.255.1 |

**Reply by bdruth** [reply]
Tried DSVPN, but can't get it to work - for some reason, it isn't saving the right port when I set it to what's in the server config file, keeps reverting back to `65401` when the server says it should be `65011`. Tried MLVPN - it seems like it's maybe less flaky, but it's "consistently slow" - avg ping of ~1s when each WAN link averages 34ms and 47ms, respectively.

Any thoughts on what else I can try? Also, what's the reason that Wireguard isn't an option for this and only an option for running MPTCP over VPN?

**Reply by bdruth** [reply]
for what it's worth, ping seems consistently low if there's no UDP traffic - if I start some traffic (e.g. video streaming) that uses UDP, then it starts getting erratic (and I think, if it's erratic enough, it starts leading to glorytun restarts?). Does any SQM get applied to `tun0`?

**Reply by Ysurac** [reply]
In col1, line2 you really ping server ip ? There is a direct route to server ip so you should get same result than a wan.
I don't understand what you mean with DSVPN and setting via server config file, all should be done via wizard (but yes there is a problem in port configuration).
Did you try glorytun UDP ?

**Reply by bdruth** [reply]
Yeah, in col1, line2 it's pinging the VPS (public) IP - ICMP goes over the VPN tunnel, afaik, at least that's why I assumed it was showing the same issue as the col2, line2 ping.

re: DSVPN - the `openmptcprouter_config.txt` saved on the VPS says that the port should be, i.e. `A Dead Simple VPN port: 65011` - but that's not what the wizard sets and if I go into the DSVPN config via LuCi directly, edit & save+apply, it doesn't seem to save right - it reverts back to the previous value. I typically do everything via the wizard, just switching to DSVPN wasn't bringing up the VPN connection, so I dug a little deeper to see what the logs were complaining about - that's where I saw the connection issues.

I believe I tried both Glorytun UDP & TCP, didn't see a difference? Sat evening I flashed back to 0.57.3 - haven't seen any similar issues since then. I have another device I can flash to 0.58.5 again to swap out and try things if that helps?

**Reply by bdruth** [reply]
Any other thoughts @Ysurac ? Should I try again w/ 58.5 or wait for an update?

**Comment by bdruth** [comment]
Traffic correlation. Light blue is tun0 traffic

<img width="670" alt="image" src="https://user-images.githubusercontent.com/444571/132109850-80772afe-d429-4fd5-b098-95847c90d76c.png">

Ping traffic, same timeframe

<img width="1171" alt="image" src="https://user-images.githubusercontent.com/444571/132109854-064756cd-51d9-435c-bbf0-b7970949a461.png">

This was a few days ago (before 0.58.5 update), similar (zoomed in to show `tun0` traffic):

<img width="666" alt="image" src="https://user-images.githubusercontent.com/444571/132110039-203b49b8-4107-421b-a9bf-540a9d3e8959.png">

Ping traffic, same timeframe

<img width="1169" alt="image" src="https://user-images.githubusercontent.com/444571/132110044-b5b9e09e-e04c-435b-a109-e3711f4aed4d.png">

**Comment by bdruth** [comment]
Just updating this in case anyone else runs into it - as of today, I'm running on the `RC` again - `openmptcprouter v0.59rc1-5.4 r0+16594-ce92de8c8c` -- and this issue is no longer happening as near as I can tell. MP scheduler: `BLEST`, Glorytun TCP, SQM (cake/piece-of-cake) on DSL, SQM autorate on Starlink.
